packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker/TimeRangeOption.tsx

```diff
@@ -16,26 +18,52 @@ const getStyles = (theme: GrafanaTheme2) => {
         cursor: pointer;
       }
     `,
-    selected: css`    
-      background: ${theme.colors.action.selected};    
-    }
-  `,
+    selected: css`
+      background: ${theme.colors.action.selected};
+      font-weight: ${theme.typography.fontWeightMedium};
+    `,
+    radio: css`
+      opacity: 0;
+
+      &:focus-visible + label {
+        ${getFocusStyles(theme)};
+      }
+    `,
+    label: css`
+      cursor: pointer;
+    `,
   };
 };
 
 interface Props {
   value: TimeOption;
   selected?: boolean;
   onSelect: (option: TimeOption) => void;
+  /**
+   *  Input identifier. This should be the same for all options in a group.
+   */
+  name: string;
 }
 
-export const TimeRangeOption = memo<Props>(({ value, onSelect, selected = false }) => {
+export const TimeRangeOption = memo<Props>(({ value, onSelect, selected = false, name }) => {
   const styles = useStyles2(getStyles);
+  // In case there are more of the same timerange in the list
+  const id = value.display + Math.floor(Math.random() * 10);
 
   return (
-    <div className={cx(styles.container, selected && styles.selected)} onClick={() => onSelect(value)} tabIndex={-1}>
-      <span>{value.display}</span>
-    </div>
+    <li className={cx(styles.container, selected && styles.selected)}>
+      <input
+        className={styles.radio}
+        checked={selected}
+        name={name}
+        type="radio"
```

Comment:
Radio groups causes automatic selection when navigating with keyboard - should we change to checkboxes or override the default behaviour?

Reply:
I noticed some behavior with the checkboxes when interacting using the screen reader. The screen reader does not identify a `checked` box immediately it's been checked. It reads it as `'not checked'`. But then, clicking the TimePicker again shows the selected option as `checked`. 

I assume this is an issue with the screen reader software. Have you encountered anything like that? @tskarhed